### PR TITLE
CORE-10453: Replace RemoteAliasLookUp with InteropIdentityLookup

### PR DESCRIPTION
**Pull request by jmacmahonr3**
An alternative more simplistic implementation of the InterOp equivalent of the MemberLookup. 

What's noteworthy about this new API is that the Application name is the sole identifier an the flow level.

### CODE REVIEW

File: interop/src/main/java/net/corda/v5/interop/InterOpIdentityInfo.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+@CordaSerializable
+public interface InterOpIdentityInfo {
+    /**
+     * @return The {@link String} identifier for the InterOp Identity.
+     */
+    @NotNull
+    String applicationName();
+
+    /**
+     * @return The {@link String} of the Identities X500.
```
Comment (yarivus) [inline interop/src/main/java/net/corda/v5/interop/InterOpIdentityInfo.java]:
```suggestion
     * @return The {@link String} of the Identity's X500.
```

**Review by yarivus — CHANGES_REQUESTED** [review]

**Review by yarivus — APPROVED** [review]
LGTM

File: interop/src/main/java/net/corda/v5/interop/InterOpIdentityInfo.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    @NotNull
+    String applicationName();
+
+    /**
+     * @return The {@link String} of the Identity's X500.
+     */
+    @NotNull
+    String getX500Name();
+
+
```
Comment (szymonsztuka) [inline interop/src/main/java/net/corda/v5/interop/InterOpIdentityInfo.java]: Remove a new line

File: interop/src/main/java/net/corda/v5/interop/InterOpIdentityInfo.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package net.corda.v5.interop;
+
+import net.corda.v5.base.annotations.CordaSerializable;
+import org.jetbrains.annotations.NotNull;
+import java.util.List;
+
+/**
+ * <p>This interface represents the alias member details like Application Name, facadeIds etc.
```
Comment (szymonsztuka) [inline interop/src/main/java/net/corda/v5/interop/InterOpIdentityInfo.java]: Very minor - I think we can remove "< p >" and extra empty line in that Javadoc

File: application/src/main/java/net/corda/v5/application/interop/InteropIdentityLookUp.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import net.corda.v5.application.interop.facade.FacadeId;
+import net.corda.v5.base.annotations.DoNotImplement;
+import net.corda.v5.base.annotations.Suspendable;
+import net.corda.v5.interop.InterOpIdentityInfo;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * <p>This interface represents the injectable service for client corDapp to look up the remote alias identity.
```
Comment (szymonsztuka) [inline application/src/main/java/net/corda/v5/application/interop/InteropIdentityLookUp.java]: We would need to clearly separate this service from any services returning normal member group, i would use explicit words like _Interoperability group_ etc. Also javadoc for other services doesn't mention `injectable`, all Corda API services are injectable.
Maybe something like that, which explicitly shows the different with existing `memberLookup`:
``The service allows a flow to retrieve alter-ego members from other networks, which are register in the holding identity's interoperability group.``

File: interop/src/main/java/net/corda/v5/interop/InterOpIdentityInfo.java (hunk 3 above)
Comment (szymonsztuka) [reply interop/src/main/java/net/corda/v5/interop/InterOpIdentityInfo.java]: Maybe a bit later not for this PR/ so just suggestion, but the description doesn't match the interface name or doesn't refer: "Identity" or "InteroOp",  how "alias member" relates to "InterOpIdentity"?

File: application/src/main/java/net/corda/v5/application/interop/InteropIdentityLookUp.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     * @return The {@link InterOpIdentityInfo} representing the alias member info for given alias x500name and host network.
+     */
+    @Suspendable
+    @NotNull
+    InterOpIdentityInfo lookup(String applicationName);
+
+    /**
+     * @param facadeId The facadeId of the facade.
+     * @return The {@link List< InterOpIdentityInfo >} representing the alias member info for given facadeId.
```
Comment (szymonsztuka) [inline application/src/main/java/net/corda/v5/application/interop/InteropIdentityLookUp.java]: Remove spaces after `<` and before `>`

File: application/src/main/java/net/corda/v5/application/interop/InteropIdentityLookUp.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * <p>This interface represents the injectable service for client corDapp to look up the remote alias identity.
+ *
+ */
+@DoNotImplement
+public interface InteropIdentityLookUp {
+
+    /**
+     * @return The {@link InterOpIdentityInfo} representing the alias member info for given alias x500name and host network.
```
Comment (szymonsztuka) [inline application/src/main/java/net/corda/v5/application/interop/InteropIdentityLookUp.java:24]: This is a crucial parameter and it's not described, is this intentional?  
What is `applicationName`, how does it relates to  "x500name" and "host network".?

**Review by szymonsztuka — COMMENTED** [review]
Needs to increment API version in gradle.properties.

File: application/src/main/java/net/corda/v5/application/interop/InteropIdentityLookUp.java (hunk 4 above)
Comment (jmacmahonr3) [reply application/src/main/java/net/corda/v5/application/interop/InteropIdentityLookUp.java]: Please check updated docs

**Review by yarivus — APPROVED** [review]
LGTM

**Review by szymonsztuka — APPROVED** [review]
Approving, the `cordaApiRevision` probably should be `7` and not `8`, but doesn't really matter.

**Pull request merged**